pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -142,6 +139,29 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 	return s.handle(&message{Msg: msg, ThreadID: thid, NextStateName: next.Name()})
 }
 
+func (s *Service) validateState(thid string, next state) error {
+	// TODO make it configurable #414
+	// for now we use defaults
+	r := retry.WithDefaults()
```

Comment:
we need retries inside the state machine? why?

Reply:
we have race condition it's possible that you receive response before update the state @talwinder50 @llorllale